manifests/dispatch_router.pp

```diff
@@ -0,0 +1,52 @@
+# == Class: capsule::dispatch_router
+#
+# Install and configure Qpid Dispatch Router
+#
+# === Parameters:
+#
+#
+class capsule::dispatch_router (
+) {
+
+  class { 'qpid::router': }
+
+  qpid::router::ssl_profile { 'router-ssl':
+    ca   => $certs::ca_cert,
+    cert => $certs::params::messaging_client_cert,
+    key  => $certs::params::messaging_client_cert,
+  }
+
+  if $capsule::pulp_master {
+    # Listen for other qpid dispatch routers
+    qpid::router::listener {'hub':
+      addr        => $capsule::qpid_router_hub_addr,
+      port        => $capsule::qpid_router_hub_port,
+      role        => 'inter-router',
+      ssl_profile => 'router-ssl',
+    }
+
+    # Connect dispatch router to the local qpid
+    qpid::router::connector { 'broker':
+      addr        => $capsule::qpid_router_broker_addr,
+      port        => $capsule::qpid_router_broker_port,
+      ssl_profile => 'router-ssl',
+      role        => 'on-demand',
+    }
+  } else {
+    qpid::router::connector { 'hub':
+      addr        => $capsule::parent_fqdn,
+      port        => $capsule::qpid_router_hub_port,
+      ssl_profile => 'router-ssl',
+      role        => 'inter-router',
+    }
+  }
+
+  # Listen for katello-agent clients
+  qpid::router::listener { 'clients':
```

Comment:
Maybe move this above the if-else so thats it not lost at the bottom given the size of the if/else ?